### PR TITLE
shell/gnrc_icmpv6_echo: print interface id in reply

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

I often find myself wondering on what interface I've received a reply with `ping6` if there is more than one.

So if there is more than one interface, print the id of the receiving interface for convenience.

```
> ping6 ff02::1
2020-02-20 18:19:38,644 # 12 bytes from fe80::d0ae:c1b:2054:58e%7: icmp_seq=0 ttl=64 rssi=-45 dBm time=4.801 ms
2020-02-20 18:19:38,652 # 12 bytes from fe80::d0ae:c1b:2054:58c%8: icmp_seq=0 ttl=64 rssi=-63 dBm time=13.422 ms (DUP!)
2020-02-20 18:19:38,661 # 12 bytes from fe80::c78:16d9:8aca:ba9a%8: icmp_seq=0 ttl=64 rssi=-44 dBm time=21.819 ms (DUP!)
2020-02-20 18:19:39,648 # 12 bytes from fe80::d0ae:c1b:2054:58e%7: icmp_seq=1 ttl=64 rssi=-44 dBm time=4.797 ms
2020-02-20 18:19:39,657 # 12 bytes from fe80::d0ae:c1b:2054:58c%8: icmp_seq=1 ttl=64 rssi=-61 dBm time=12.917 ms (DUP!)
2020-02-20 18:19:39,666 # 12 bytes from fe80::c78:16d9:8aca:ba9a%8: icmp_seq=1 ttl=64 rssi=-43 dBm time=21.679 ms (DUP!)
2020-02-20 18:19:40,657 # 12 bytes from fe80::d0ae:c1b:2054:58e%7: icmp_seq=2 ttl=64 rssi=-41 dBm time=4.795 ms
```

On Linux, `ping` only seems to do that for link-local addresses, but the information seems helpful for global addresses too.

### Testing procedure

Run `ping6`.

### Issues/PRs references

none

### CODE REVIEW

**Comment by smlng** [comment]
> On Linux, `ping` only seems to do that for link-local addresses, but the information seems helpful for global addresses too.

well, for global addresses it shouldn't be necessary bc you can infer the interface from the IP prefix, which must be unique per interface. Which you cannot for link local addresses bc they all have `fe80::` prefix.

From a tooling perspective it might be nice to match RIOTs output with that of Linux `ping6`. So one could use the same parser, e.g. for automated tests.

**Comment by benpicco** [comment]
> for global addresses it shouldn't be necessary bc you can infer the interface from the IP prefix, which must be unique per interface. 

I know, I just think it can be handy for debugging to always see over which interface the reply was received

**Comment by miri64** [comment]
I agree with @smlng, that for a global address the interface should not be appended (I'm not even sure this is legal... [RFC 5952](https://tools.ietf.org/html/rfc5952) doesn't mention this format so no idea where it is specified :-/).

File: sys/shell/commands/sc_gnrc_icmpv6_echo.c [hunk 1]
```diff
@@ -364,8 +367,16 @@ static void _print_reply(_ping_data_t *data, gnrc_pktsnip_t *icmpv6,
             data->num_recv++;
             dupmsg += 7;
         }
-        printf("%u bytes from %s: icmp_seq=%u ttl=%u", (unsigned)icmpv6->size,
-               from_str, recv_seq, hoplimit);
+        if (GNRC_NETIF_NUMOF == 1 || if_pid == 0) {
```
Comment (miri64) [inline sys/shell/commands/sc_gnrc_icmpv6_echo.c]: parenthesis missing.

Comment (miri64) [inline sys/shell/commands/sc_gnrc_icmpv6_echo.c]: You just need to add a `ipv6_addr_is_link_local(from)` here to have global addresses not be outputted with an interface.

File: sys/shell/commands/sc_gnrc_icmpv6_echo.c [hunk 2]
```diff
@@ -322,10 +322,13 @@ static void _pinger(_ping_data_t *data)
 
 static void _print_reply(_ping_data_t *data, gnrc_pktsnip_t *icmpv6,
                          ipv6_addr_t *from, unsigned hoplimit,
-                         int16_t rssi)
+                         gnrc_netif_hdr_t *netif_hdr)
 {
     icmpv6_echo_t *icmpv6_hdr = icmpv6->data;
 
+    kernel_pid_t if_pid = netif_hdr ? netif_hdr->if_pid : 0;
```
Comment (miri64) [inline sys/shell/commands/sc_gnrc_icmpv6_echo.c]:
```suggestion
    kernel_pid_t if_pid = netif_hdr ? netif_hdr->if_pid : KERNEL_PID_UNDEF;
```

File: sys/shell/commands/sc_gnrc_icmpv6_echo.c [hunk 3]
```diff
@@ -364,8 +367,17 @@ static void _print_reply(_ping_data_t *data, gnrc_pktsnip_t *icmpv6,
             data->num_recv++;
             dupmsg += 7;
         }
-        printf("%u bytes from %s: icmp_seq=%u ttl=%u", (unsigned)icmpv6->size,
-               from_str, recv_seq, hoplimit);
+        if ((GNRC_NETIF_NUMOF == 1) || (if_pid == 0) ||
```
Comment (miri64) [inline sys/shell/commands/sc_gnrc_icmpv6_echo.c]:
```suggestion
        if ((GNRC_NETIF_NUMOF == 1) || (if_pid == KERNEL_PID_UNDEF) ||
```

**Pull request merged**